tutorials/editor_basics/coordinates_system.md

```diff
@@ -23,123 +34,165 @@ public class EditorAreaIllustration extends AnAction {
         CaretModel caretModel = editor.getCaretModel();
     }
     @Override
-    public void update(AnActionEvent e) {
-       //...
-    }
+    public void update(AnActionEvent e) { /* ... */ }
 }
 ```
 
-## 2.3. Logical position
+## Caret Coordinate Systems
+When a `Document` is opened the `Editor` assigns an internal, zero-based coordinate system to lines and columns in the `Document`.
+The first line in a `Document` and the first character in each line are assigned the zero position.
+Note that the `Editor` coordinate system is different from what is shown in the editor UI, which is one-based rather than zero-based.
+
+The [Multiple Carets](/reference_guide/multiple_carets.md) documentation covers the subject of more than one caret in a document.
+For the purposes of this tutorial, be aware there may be more than one caret in an `Document` at any given time.
+Consequently, examples will be based on using the _Primary Caret_ in a `Document`.
+If there is only one caret in a `Document`, it is the Primary Caret.
+For the case of multiple carets in a `Document`, the Primary Caret is the one on which query and update methods in the model operate at the moment.
+
+### Caret Logical Position
+The caret _Logical Position_ is a zero-based, (line and column) position of the caret in the Editor Tool Window.
```

Comment:
Editor != "Tool Window" - that's what we call tabs located on left,bottom,right stripes of IDE main window. please note further occurrences below in this page

Reply:
Trying to relate effects to what users see. Obvious FAIL. Removed.